### DISCUSSION TITLE
Adding r'''.''' to "tokenizer.suffix_search" breaks English tokenization

### DISCUSSION THREAD

**Discussion by bentherien** [explosion/spaCy#7303] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour

```
import spacy

nlp = spacy.load("en_core_web_lg") # same output for 'en_core_sci_scibert' and 'en_core_sci_lg'

example = "The maximum temperature varies from 10,000 K \
to 13,200 K."

doc = nlp(example)
print([token.text for token in doc])
print("\n\nHere \"K.\" should be \"K\",\".\"\n\n")

suffixes = nlp.Defaults.suffixes + [r'''.''']
suffix_regex = spacy.util.compile_suffix_regex(suffixes)
nlp.tokenizer.suffix_search = suffix_regex.search

doc = nlp(example)
print([token.text for token in doc])

## Output 
['The', 'maximum', 'temperature', 'varies', 'from', '10,000', 'K', 'to', '13,200', 'K.']
```

Here "K." should be "K","."

```
['T', 'h', 'e', 'm', 'a', 'x', 'i', 'm', 'u', 'm', 't', 'em', 'p', 'e', 'r', 'a', 't', 'u', 'r', 'e', 'v', 'a', 'r', 'i', 'e', 's', 'f', 'r', 'o', 'm', '1', '0', ',', '0', '0', '0', 'K', 't', 'o', '1', '3', ',', '2', '0', '0', 'K', '.']
```

## Your Environment

* Operating System: Ubuntu 20.10
* Python Version Used: Python 3.8.6
* spaCy Version Used: spaCy v3.0.3
* Environment Information: pip freeze > requirements.txt (output attached)
[requirements.txt](https://github.com/explosion/spaCy/files/6063663/requirements.txt)

## More on the issue
I am working with scientific text and would like to find a way to parse off periods. In the above example, "K." is not tokenized into "K","." using spacy's default tokenizer. Adding "." to suffixes breaks the tokenization. How else could I do this?

**Comment by polm** [comment, marked as answer]
The issue here is that the `suffixes` is a regular expression, and in a regular expression an ordinary period means "any character". If I change your code to be like the below it should work the way you want it to:

    suffixes = nlp.Defaults.suffixes + [r"\."]

**Comment by svlandeg** [comment]
(Maintenance note: this question is well suited for the discussions forum, so I'm moving it there. While it looks like this issue gets closed, in fact the linked discussion thread will remain open to continue the conversation)
